File: client/scala-gatling/src/gatling/scala/org/openapitools/client/model/Keyword.scala

package org.openapitools.client.model


case class Keyword (
    /* Name of the keyword */
    _keywordName: Option[String],
    /* Key of the keyword (short code) */
    _keywordKey: Option[String],
    /* Internal keyword ID */
    _keywordId: Option[Integer],
    /* Date when this keyword was added */
    _dateAdded: Option[String]
)
object Keyword {
    def toStringBody(var_keywordName: Object, var_keywordKey: Object, var_keywordId: Object, var_dateAdded: Object) =
        s"""
        | {
        | "keywordName":$var_keywordName,"keywordKey":$var_keywordKey,"keywordId":$var_keywordId,"dateAdded":$var_dateAdded
        | }
        """.stripMargin
}

File: models/Keyword.js
const utils = require('../utils/utils');

module.exports = {
    fields: (prefix = '', isInput = true, isArrayChild = false) => {
        const {keyPrefix, labelPrefix} = utils.buildKeyAndLabel(prefix, isInput, isArrayChild)
        return [
            {
                key: `${keyPrefix}keywordName`,
                label: `Name of the keyword - [${labelPrefix}keywordName]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}keywordKey`,
                label: `Key of the keyword (short code) - [${labelPrefix}keywordKey]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}keywordId`,
                label: `Internal keyword ID - [${labelPrefix}keywordId]`,
                type: 'integer',
            },
            {
                key: `${keyPrefix}dateAdded`,
                label: `Date when this keyword was added - [${labelPrefix}dateAdded]`,
                type: 'string',
            },
        ]
    },
    mapping: (bundle, prefix = '') => {
        const {keyPrefix} = utils.buildKeyAndLabel(prefix)
        return {
            'keywordName': bundle.inputData?.[`${keyPrefix}keywordName`],
            'keywordKey': bundle.inputData?.[`${keyPrefix}keywordKey`],
            'keywordId': bundle.inputData?.[`${keyPrefix}keywordId`],
            'dateAdded': bundle.inputData?.[`${keyPrefix}dateAdded`],
        }
    },
}
